### ISSUE TITLE
`exec` command should force the last interactive command to history

### ISSUE THREAD

**Issue by ahti** [fish-shell/fish-shell#907]
I have a function to exec ssh to a server:

```
function essh
   if [ "$argv" = "" ]
       echo "Need some arguments, right?"
       return
   end
   exec ssh $argv
end
```

Fish does not provide me with auto completion when typing essh, which does make it harder to use. I suspect fish saves things to the history after the command was run. Maybe this could be changed?

**Comment by krader1961** [comment]
If the shell is interactive the `exec` builtin should write the last interactive command that was entered to history before replacing the shell with the specified program.

**Comment by ridiculousfish** [comment]
This got fixed a long time ago as ded81ec186 (though you wouldn't know it by the commit message, ugh)
